plugin/builtin/namespacetransformer/NamespaceTransformer.go

```diff
@@ -53,6 +53,7 @@ func (p *plugin) Transform(m resmap.ResMap) error {
 		}
 	}
 	p.updateClusterRoleBinding(m)
+	p.updateServiceReference(m)
```

Comment:
Looks good for now. We can think about making this kind of update generic. Maybe a new transformer or extend current namereference transformer. When both `name` and `namespace` is needed to be set or updated at the same time, the transformer can handle both of them.